common/middleware/instrument.go

```diff
@@ -39,14 +40,32 @@ func (i Instrument) Wrap(next http.Handler) http.Handler {
 	})
 }
 
+var invalidChars = regexp.MustCompile(`[^a-zA-Z0-9]+`)
+
 func (i Instrument) getRouteName(r *http.Request) string {
 	var routeMatch mux.RouteMatch
 	if !i.RouteMatcher.Match(r, &routeMatch) {
-		return "unmatched_path"
+		return MakeLabelValue(r.URL.Path)
 	}
 	name := routeMatch.Route.GetName()
 	if name == "" {
-		return "unnamed_path"
+		return MakeLabelValue(r.URL.Path)
 	}
 	return name
 }
+
+// MakeLabelValue converts a Gorilla mux path to a string suitable for use in
+// a Prometheus label value.
+func MakeLabelValue(path string) string {
```

Comment:
for testing; we have a weird convention when tests are in a different package, so things you want to test need to be to be public.  We should change that, but not today.